### PR TITLE
Updated travis.yml for micromamba

### LINKED ISSUE #423: switch CI to Micromamba
### 🎯 Todo

Conda environment became sluggish, and it is even causing failures in the Continuus-integration tests : the conda environemnt crashes during build . https://app.travis-ci.com/github/radis/radis/jobs/562227307  (fails #422) 

### 💡 Possible solutions

We could use Mamba instead of Conda, which is known to be faster but also much more memory efficient. Mamba requires Conda, but Micromamba doesn't :  

https://labs.epi2me.io/conda-or-mamba-for-production/

https://mamba.readthedocs.io/en/latest/installation.html

Good First issue : help us replace Conda with Mamba in the [.travis-ci.yml](https://github.com/radis/radis/blob/develop/.travis.yml) conifguration file 

---
<sub>*Tip : you can vote for [New Features](https://github.com/radis/radis/issues/79) on https://feathub.com/radis/radis*</sub>

### PR DESCRIPTION
**Pull request by Supriya1702**
### Description

This pull request replaces miniconda with micromamba in `.travis.yml`. 
@erwanp @anandxkumar I ran the changed code on my system it passed the tests but got stuck at 8th test of `test_examples.py` same happens when I use conda environment, I wonder what the issue is.

 

Fixes #423

### CODE REVIEW

**Comment by anandxkumar** [comment]
Great work @Supriya1702, the Travis has been occupied by another pull request, so we will have to wait till it gets terminated to see if your changes are working.

File: .travis.yml [hunk 1]
```diff
@@ -20,25 +20,22 @@ jobs:
...
+        - source ~/.bashrc
         # Useful for debugging any issues with conda
-        - conda info -a
-
+        - micromamba info 
         # Update python version in environment
         - sed -i -E 's/(python=)(.*)/\1'$TRAVIS_PYTHON_VERSION'/' ./environment.yml
         # Create conda environment
-        - conda env create -n radis-env -f environment.yml
+        - micromamba install xtensor -c conda-forge
```
Comment (anandxkumar) [inline .travis.yml:34]: Add `-y` argument, in the end, to  select yes automatically when `y/n` is prompted

**Review by anandxkumar — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -20,25 +20,22 @@ jobs:
...
+        - micromamba create -f environment.yml
```
Comment (anandxkumar) [inline .travis.yml:35]: Same here, add `-y` at the end

**Comment by Supriya1702** [comment]
Hello @erwanp @anandxkumar , seems like the micromamba environment got created on `travis-ci`. Some of the tests failed and all have the same issue `micromamba/envs/radis-env/lib/python3.8/site-packages/pandas/core/indexes/base.py:5637: InvalidIndexError`. What could be possible reasons? I believe it is due to the installation of a specific version of pandas package by micromamba.

**Comment by arunavabasu-03** [comment]
hey @Supriya1702, please check **[this issue ](https://github.com/pytest-dev/pytest-qt/issues/206)**, I think this should be relevant. 😅

**Comment by erwanp** [comment]
Hello @Supriya1702 , well done ! 

The errors are not related to your changes, I think it's related to some new Pandas version . I'll look it up

File: .travis.yml (hunk 1 above)
Comment (anandxkumar) [reply .travis.yml]: Great!

**Review by anandxkumar — APPROVED** [review]

**Comment by anandxkumar** [comment]
I tested locally with changes [here](https://github.com/radis/radis/pull/428). I believe it should pass if Travis passes in PR [#428](https://github.com/radis/radis/pull/428) with some updates in cython library using anaconda.

**Comment by erwanp** [comment]
Tests pass except from a tiny trailing-space formatting test; I'm merging. 

Congratulatinos @Supriya1702 for this first contribution !

File: .travis.yml [hunk 3]
```diff
@@ -20,25 +20,23 @@ jobs:
...
+        - micromamba activate base
         # Useful for debugging any issues with conda
-        - conda info -a
-
+        - micromamba info
         # Update python version in environment
         - sed -i -E 's/(python=)(.*)/\1'$TRAVIS_PYTHON_VERSION'/' ./environment.yml
         # Create conda environment
-        - conda env create -n radis-env -f environment.yml
+        - micromamba install xtensor -c conda-forge -y
```
Comment (erwanp) [inline .travis.yml:35]: @Supriya1702 why was Xtensor added here?

Comment (Supriya1702) [reply .travis.yml:35]: @erwanp, In order to install the dependencies of radis mentioned in environment.yml we need to install some prerequisites based in the xtensor library of micromamba, else the dependencies won't install and the build will fail

Comment (erwanp) [reply .travis.yml:35]: We don't use Xtensor internally in Radis, so it's probably a radis dependency that requires it. Isn't Mamba suppose to find this out by itself?

Comment (Supriya1702) [reply .travis.yml:35]: I tried without installing the xtensor, upon creating the virtual environment it was showing errors that some important library missing so can't install dependencies. xtensor contains some important c++ based libraries for GCC compiler:
Following are dependencies in  xtensor:
xtensor == 0.21.5
  ├─ libgcc-ng [>=7.3.0]
  │ ├─ _libgcc_mutex [0.1 conda_forge]
  │ └─ _openmp_mutex [>=4.5]
  │   ├─ _libgcc_mutex already visited
  │   └─ libgomp [>=7.3.0]
  │     └─ _libgcc_mutex already visited
  ├─ libstdcxx-ng [>=7.3.0]
  └─ xtl [>=0.6.9,<0.7]
    ├─ libgcc-ng already visited
    └─ libstdcxx-ng already visited

I believe these libraries are essential for installing basic libraries in micromamba, which can be inferred from the fact that it is very light weight so some dependencies don't come in the base environment.

Also, why has the recent Travis build failed, I tested locally it was working fine. [build][https://app.travis-ci.com/github/radis/radis/jobs/570208518]
File "/home/travis/micromamba/envs/radis-env/bin/pytest", line 8 in <module>
Aborted
The command "xvfb-run pytest --cov=./" exited with 134.

Reference: [https://github.com/mamba-org/mamba]

Comment (Supriya1702) [reply .travis.yml:35]: I tried without installing the xtensor, upon creating the virtual environment it was showing errors that some important library missing so can't install dependencies. xtensor contains some important c++ based libraries for GCC compiler:
Following are dependencies in  xtensor:
xtensor == 0.21.5
  ├─ libgcc-ng [>=7.3.0]
  │ ├─ _libgcc_mutex [0.1 conda_forge]
  │ └─ _openmp_mutex [>=4.5]
  │   ├─ _libgcc_mutex already visited
  │   └─ libgomp [>=7.3.0]
  │     └─ _libgcc_mutex already visited
  ├─ libstdcxx-ng [>=7.3.0]
  └─ xtl [>=0.6.9,<0.7]
    ├─ libgcc-ng already visited
    └─ libstdcxx-ng already visited

I believe these libraries are essential for installing basic libraries in micromamba, which can be inferred from the fact that it is very light weight so some dependencies don't come in the base environment.

Also, why has the recent Travis build failed, I tested locally it was working fine. [build](https://app.travis-ci.com/github/radis/radis/jobs/570208518)
File "/home/travis/micromamba/envs/radis-env/bin/pytest", line 8 in <module>
Aborted
The command "xvfb-run pytest --cov=./" exited with 134.
[Reference](https://github.com/mamba-org/mamba)

Comment (erwanp) [reply .travis.yml:35]: Ok thank you ! 

for local Travis build --> more on Slack !

**Pull request merged**